src/Swarm/TUI/Model.hs

```diff
@@ -691,6 +690,50 @@ promptUpdateL = lens g s
     CmdPrompt _ _ -> mkReplForm $ mkCmdPrompt inputText
     SearchPrompt _ _ -> mkReplForm $ SearchPrompt inputText (ui ^. uiReplHistory)
 
+-- ----------------------------------------------------------------------------
+--                                Runtime state                              --
+-- ----------------------------------------------------------------------------
+
+data RuntimeState = RuntimeState
+  { _webPort :: Maybe Port
+  , _upstreamRelease :: Either NewReleaseFailure String
+  , _eventLog :: Seq LogEntry
+  }
+
+initRuntimeState :: RuntimeState
+initRuntimeState =
+  RuntimeState
+    { _webPort = Nothing
+    , _upstreamRelease = Left NoUpstreamRelease
+    , _eventLog = mempty
+    }
+
+makeLensesWith (lensRules & generateSignatures .~ False) ''RuntimeState
+
+-- | The port on which the HTTP debug service is running.
+webPort :: Lens' RuntimeState (Maybe Port)
+
+-- | The upstream release version.
+upstreamRelease :: Lens' RuntimeState (Either NewReleaseFailure String)
+
+-- | A log of runtime events.
+--
+-- This logging is separate from the logging done during game-play.
+-- If some error happens before a game is even selected, this is the
+-- place to log it.
+eventLog :: Lens' RuntimeState (Seq LogEntry)
+
+-- ----------------------------------------------------------------------------
+--                                   APPSTATE                                --
+-- ----------------------------------------------------------------------------
+
+-- | The 'AppState' just stores together the game state and UI state.
```

Comment:
Yep, it's still just a collection of other states, so `AppState` has not gained any status. :smirk: 

I was considering putting the app-wide stuff in `AppState`, but it would look weird to have `Maybe Int` on the same level as `GameState` and I already got used to the `s ^. uiState . uiSomeLens` idiom.